yaml-tests/src/test/resources/documentation-queries/between-operator-queries.yamsql

```diff
@@ -0,0 +1,71 @@
+---
+options:
+  supported_version: 4.3.5.0
+---
+schema_template:
+    create table products(id bigint, name string, category string, price bigint, stock integer, primary key(id))
+    create index price_idx as select price from products order by price
+    create index category_idx as select category from products order by category
+    create index name_idx as select name from products order by name
+---
+setup:
+  steps:
+    - query: insert into products
+            values (1, 'Widget A', 'Electronics', 100, 50),
+                   (2, 'Widget B', 'Electronics', 150, 30),
+                   (3, 'Gadget X', 'Electronics', 200, 20),
+                   (4, 'Tool A', 'Hardware', 80, 100),
+                   (5, 'Tool B', 'Hardware', 120, 15),
+                   (6, 'Book A', 'Media', 25, 200),
+                   (7, 'Book B', 'Media', 30, 150)
+---
+test_block:
+  name: between-operator-tests
+  preset: single_repetition_ordered
+  tests:
+    # BETWEEN with numbers
+    -
+      - query: SELECT name, price
+              FROM products
+              WHERE price BETWEEN 100 AND 150
+              ORDER BY price
+      - supported_version: 4.3.5.0
```

Comment:
This is probably redundant in light of having a file-level `supported_version` option